homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,485 @@
+        icon="mdi:solar-power",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_consumption_solar",
+        translation_key="power_consumption_solar",
+        icon="mdi:solar-power",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="energy_consumed_battery",
+        translation_key="energy_consumed_battery",
+        icon="mdi:home-battery",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_consumption_battery",
+        translation_key="power_consumption_battery",
+        icon="mdi:home-battery",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="energy_consumed_grid",
+        translation_key="energy_consumed_grid",
+        icon="mdi:transmission-tower",
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_consumption_grid",
+        translation_key="power_consumption_grid",
+        icon="mdi:transmission-tower",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_registry_enabled_default=False,
+    ),
+]
+SENSOR_DESCRIPTIONS = {
+    "SOLAR_PV": [
+        SensorEntityDescription(
+            key="energy_supplied",
+            translation_key="solar_pv_energy_supplied",
+            icon="mdi:solar-power",
+            state_class=SensorStateClass.TOTAL,
+            native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+        ),
+        SensorEntityDescription(
+            key="power_supply",
+            translation_key="solar_pv_power_supply",
+            icon="mdi:solar-power",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+            device_class=SensorDeviceClass.POWER,
+        ),
+    ],
+    "WATER_HEATER": (
+        SENSOR_DESCRIPTIONS_GENERIC_CONSUMER
+        + [
+            SensorEntityDescription(
+                key="available_energy",
```

Comment:
All the keys of the water heater are not correct. Please review them and correct them.

Reply:
These are correct.

Reply:
The keys are used to map to entries in the combined energy data.

Reply:
I would suggest something like `available_liters` as the unit is liters. I can't imagine how energy can be described in liters

Reply:
I can't change this, this is how it is represented in the API. The Unit is correct.